kernel/src/syscall.rs

```diff
@@ -648,4 +648,12 @@ pub trait UserspaceKernelBoundary {
         state: &Self::StoredState,
         writer: &mut dyn Write,
     );
+
+    /// Store architecture specific (e.g. CPU registers or status flags) data
+    /// for a process. On success returns the number of elements written to out.
+    fn store_context(
+        &self,
+        state: &Self::StoredState,
+        out: &mut [usize],
```

Comment:
Can we change this to use a slice of `u8`s instead? It's unconventional to pass around slices of `usize` in the kernel and thus might make interop with other interfaces more tricky. Furthermore, using `usize` can be tricky w.r.t. endianness. While platforms are free to choose their encoding of the `StoredState` structs (after all, they won't ever be compatible to each other), it's at least important that the header fields can be read for other platforms as well and thus use a fixed endianness. I don't expect any (significant) overhead caused by using `usize::{from,to}_le_bytes` etc.